app/packages/viz3d/src/draco_loader.ts

```diff
@@ -0,0 +1,186 @@
+import * as three from "three"
+import * as draco from "draco3d"
+
+import * as draco_worker_lib from "./draco_worker"
+import draco_worker from "./draco_worker?worker"
+
+import * as worker_util from "./worker_util"
+export {LogLevel} from "./worker_util"
+
+//export { DracoTaskConfig } from "./draco_worker"
+
+// TODO: Improve performance.
+// As of Sun Feb 6 14:15:05 PST 2022 this can
+// load ~5mb point clouds (compressed to ~650kb) in ~65ms/each. 
```

Comment:
Yes, 65ms is too slow for the main thread. In a web worker this is fine.

Reply:
Well I guess this will depend on what the needs of the typical user are. If they are regularly dealing with 2gb point clouds, there is no helping them.

And yes, most of that 65ms is happening on the worker side, but there is still a noticeable delay in thumbnail generation while scrolling through the samples.